spec/views/hq/users/index_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+describe "hq/users/index.html.erb", type: :view do
+  it 'should display a valid user' do
+    assign(:users, [ FactoryGirl.build_stubbed(:user, user_name: 'User1', email: 'Email1@example.com') ] )
+    render 
+    expect(rendered).to match /User1/
+    expect(rendered).to match /Email1@example.com/
+  end
+
+  it 'should have a New User link' do
+    assign(:users, [])
+    render
+    expect(rendered).to have_link('New User', new_hq_user_path)
+  end
+
+  describe 'should display assigned active sponsors correctly' do
+
+    context 'no active sponsors' do
+      it 'displays 0 active sponsors' do
+        @users = [ instance_double(User, active_sponsors: [], user_name: 'usr', email: 'email@e.e') ]
+        render
+        expect(rendered).to have_selector('td', text: '0')
```

Comment:
@robbiejackson Having given this more thought after our conversation, on the one hand I think that this is still unnecessarily overstepping the boundaries of a focused unit test. The point is to test the view's successful rendering of data it is given, and not of any underlying logic since that is tested elsewhere. With that in mind, 'Assigned Active Sponsors' is no different than other attributes of the instance variable passed to the view and does not need to be handled specially. To test the rendered page, then, all we need is
```ruby
it 'should display users' do
  user = instance_double User, :user_name => 'H. Vetinari', 
                               :email => 'vetinari@a-m.gov', 
                               :active_sponsors => (1..5).to_a
  assigns :users, [user]

  render

  expect(rendered).to match /H\. Vetinari/
  expect(rendered).to match /vetinari@a-m\.gov/
  expect(rendered).to match '5'
end
```

On the other hand, if my opinion is not shared by the rest of the team and the test is kept as is, I would recommend making it precise and ensuring that the sponsors' count is displayed in the correct table column, not simply in a `'td'`.